src/ArborX_LinearBVH.hpp

```diff
@@ -55,39 +55,27 @@ class BoundingVolumeHierarchy
 
   bounding_volume_type bounds() const noexcept { return _bounds; }
 
-  template <typename ExecutionSpace, typename Predicates, typename... Args>
+  template <typename ExecutionSpace, typename Predicates,
+            typename CallbackOrView, typename View, typename... Args>
   void query(ExecutionSpace const &space, Predicates const &predicates,
+             CallbackOrView &&callback_or_view, View &&view,
              Args &&... args) const
   {
+    Details::check_valid_access_traits(Traits::PredicatesTag{}, predicates);
     using Access = Traits::Access<Predicates, Traits::PredicatesTag>;
-    static_assert(
-        Details::is_complete<Access>::value,
-        "Must specialize 'Traits::Access<Predicates,Traits::PredicatesTag>'");
-    static_assert(
-        Details::has_memory_space<Access>::value,
-        "Traits::Access<Predicates,Traits::PredicatesTag> must define "
-        "'memory_space' member type that is a valid Kokkos memory space");
-    static_assert(
-        KokkosExt::is_accessible_from<typename Access::memory_space,
-                                      ExecutionSpace>::value,
-        "Traits::Access<Predicates,Traits::PredicatesTag>::memory_space must "
-        "be accessible from the bounding volume hierarchy execution space");
-    static_assert(
-        Details::has_size<Access>::value,
-        "Traits::Access<Predicates,Traits::PredicatesTag> must define "
-        "'size()' member function");
-    static_assert(
-        Details::has_get<Access>::value,
-        "Traits::Access<Predicates,Traits::PredicatesTag> must define 'get()' "
-        "member function");
-    using Tag =
-        typename Details::Tag<Details::decay_result_of_get_t<Access>>::type;
-    static_assert(std::is_same<Tag, Details::NearestPredicateTag>::value ||
-                      std::is_same<Tag, Details::SpatialPredicateTag>::value,
-                  "Invalid tag for the predicates");
+    static_assert(KokkosExt::is_accessible_from<typename Access::memory_space,
+                                                ExecutionSpace>::value,
+                  "Predicates must be accessible from the execution space");
```

Comment:
Shouldn't this be part of `check_valid_access_traits`?

Reply:
The access traits does not know anything about the execution space.  You would have to pass it and I am reluctant to do that because I think introspection of the access traits should not care about where it is accessed from.

Reply:
Strongly disagree on that one